kasa/tests/test_klapprotocol.py

```diff
@@ -400,24 +400,35 @@ async def _return_response(url: URL, params=None, data=None, *_, **__):
 
 
 @pytest.mark.parametrize(
-    "response_status, expectation",
+    "response_status, credentials_match, expectation",
     [
-        ((403, 403, 403), pytest.raises(AuthenticationException)),
-        ((200, 403, 403), pytest.raises(AuthenticationException)),
-        ((200, 200, 403), pytest.raises(AuthenticationException)),
-        ((200, 200, 400), pytest.raises(SmartDeviceException)),
+        ((403, 403, 403), True, pytest.raises(SmartDeviceException)),
+        ((200, 403, 403), True, pytest.raises(SmartDeviceException)),
+        ((200, 200, 403), True, pytest.raises(AuthenticationException)),
+        ((200, 200, 400), True, pytest.raises(SmartDeviceException)),
+        ((200, 200, 200), False, pytest.raises(AuthenticationException)),
```

Comment:
Would you mind defining these with `pytest.param(..., id="test-something-something)` to make it easier to follow what is being tested?

Reply:
Done